collated/main/A0139920A.md

```diff
@@ -0,0 +1,297 @@
+            return new CommandResult(Messages.MESSAGE_INVALID_TASK_DISPLAYED_INDEX);
+        }
+
+        ReadOnlyTask taskToDone = lastShownList.get(targetIndex - 1);
+
+        try {
+            model.doneTask(taskToDone, dataType, targetIndex);
+        } catch (TaskNotFoundException pnfe) {
+            assert false : "The target task cannot be missing";
+        }
+
+        return new CommandResult(String.format(MESSAGE_DONE_TASK_SUCCESS, taskToDone));
+    }
+
+}
+```
+###### /java/seedu/todoList/logic/commands/EditCommand.java
+``` java
+    public EditCommand(String name, String date, String endDate, String priority, int targetIndex, String dataType)
+            throws IllegalValueException {
+    	this.targetIndex = targetIndex;
+    	this.dataType = dataType;
+        this.toEdit = new Todo(
+                new Name(name),
+                new StartDate(date),
+                new EndDate(endDate),
+                new Priority(priority),
+                "false"
+        );
+    }
+    
+    /**
+     * Edit Event
+     * Convenience constructor using raw values.
+     * @throws IllegalValueException if any of the raw values are invalid
+     */
+```
+###### /java/seedu/todoList/logic/commands/EditCommand.java
+``` java
+    public EditCommand(String name, String date, String endDate, String startTime, String endTime, int targetIndex, String dataType)
+            throws IllegalValueException {
+    	this.targetIndex = targetIndex;
+    	this.dataType = dataType;
+        this.toEdit = new Event(
+                new Name(name),
+                new StartDate(date),
+                new EndDate(endDate),
+                new StartTime(startTime),
+                new EndTime(endTime),
+                "false"
+        );
+    }
+    
+    /**
+     * Edit Deadline
+     * Convenience constructor using raw values.
+     * @throws IllegalValueException if any of the raw values are invalid
+     */
+    public EditCommand(String name, String date, String endTime, int targetIndex, String dataType)
+            throws IllegalValueException {
+    	this.targetIndex = targetIndex;
+    	this.dataType = dataType;
+        this.toEdit = new Deadline(
+                new Name(name),
+                new StartDate(date),
+                new EndTime(endTime),
+                "false"
+        );
+    }
+    
+    /**
+     * Constructor for undo
+     */
+```
+###### /java/seedu/todoList/logic/commands/EditCommand.java
+``` java
+    public EditCommand(ReadOnlyTask original, String dataType, ReadOnlyTask toEdit) {
+    	this.taskToEdit = original;
+    	this.toEdit = (Task) toEdit;
+    	this.targetIndex = -1;
+    	this.dataType = dataType;
+    }
+
+```
+###### /java/seedu/todoList/logic/commands/EditCommand.java
+``` java
+    @Override
+    public CommandResult execute() {
+    	if(this.taskToEdit == null && this.targetIndex != -1) {
+	    	UnmodifiableObservableList<ReadOnlyTask> lastShownList = null;
```

Comment:
i guess you are using tabs for indentation; the coding standard requires using 4 spaces for indentation